tchannel-core/src/main/java/com/uber/tchannel/messages/EncodedRequest.java

```diff
@@ -69,6 +70,14 @@ public String getHeader(String key) {
         return getHeaders().get(key);
     }
 
+    public void updateHeaders(Map<String, String> headers) {
```

Comment:
Nit: I would call this setHeaders to make it clear that existing headers won't be retained